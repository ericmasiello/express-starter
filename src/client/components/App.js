/* @flow */
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router';
import styles from './App.scss';

// Note: Top level components must be class based for HMR to function
// eslint-disable-next-line react/prefer-stateless-function
export default class App extends Component {
  render() {
    const { children } = this.props;
    return (
      <div>
        <header className={styles.hero}>
          <h2>Universal React Express Starter App</h2>
        </header>
        <nav className={styles.nav}>
          <Link to="/">Home</Link>
          <Link to="detail/1234">Details</Link>
        </nav>
        <div>
          {children}
        </div>
      </div>
    );
  }
}

App.propTypes = {
  children: PropTypes.node.isRequired,
};
