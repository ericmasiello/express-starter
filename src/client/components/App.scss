@import '../styles/vars';

.hero {
  padding: 40px;
  background-image: url('../images/background.jpg');
  background-size: cover;
  background-position: center;
  color: $hero-color;
  min-height: 250px;
  display: flex;
  justify-content: center;
  align-items: center;
  text-shadow: 1px 0 1px rgba(0, 0, 0, 0.5);
  font-size: 2rem;
}

.nav {
  display: flex;
  padding: 1rem;

  > *:not(:first-child) {
    padding-left: 0.5rem;
  }
}