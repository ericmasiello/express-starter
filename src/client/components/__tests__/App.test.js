import React from 'react';
import renderer from 'react-test-renderer';
import App from '../App';

it('should render', () => {
  const component = renderer.create(
    <App>
      Hello World!
    </App>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
