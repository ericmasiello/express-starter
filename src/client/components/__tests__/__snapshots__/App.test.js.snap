// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<div>
  <header
    className={undefined}
  >
    <h2>
      Universal React Express Starter App
    </h2>
  </header>
  <nav
    className={undefined}
  >
    <a
      onClick={[Function]}
      style={Object {}}
    >
      Home
    </a>
    <a
      onClick={[Function]}
      style={Object {}}
    >
      Details
    </a>
  </nav>
  <div>
    Hello World!
  </div>
</div>
`;
